controllers/topology/reconcile_state.go

```diff
@@ -347,6 +457,13 @@ func (r *ClusterReconciler) updateMachineDeployment(ctx context.Context, cluster
 		return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: currentMD.Object})
 	}
 
+	// Patch MachineHealthCheck for the MachineDeployment.
+	if desiredMD.MachineHealthCheck != nil || currentMD.MachineHealthCheck != nil {
```

Comment:
What if the user dropped the MHC config from the MDTopology? Shouldn't the MHC be deleted then?

Reply:
If desiredMD is nil and currentMD in not nil here this check enters into reconcileMachineHealthCheck and correctly deletes an existing healthCheck - am I understanding you right?

Reply:
Sorry I think I misread the code. Looks fine to me now :)